pennylane/templates/embeddings/qaoaembedding.py

```diff
@@ -262,14 +277,16 @@ def compute_decomposition(
                 op_list.append(local_field(weights[l][2], wires=wires[1:2]))
 
             else:
-                for i in range(len(wires)):
-                    op_list.append(
-                        qml.MultiRZ(
-                            weights[l][i], wires=wires.subset([i, i + 1], periodic_boundary=True)
-                        )
+                op_list.extend(
+                    qml.MultiRZ(
+                        weights[l][i], wires=wires.subset([i, i + 1], periodic_boundary=True)
                     )
-                for i in range(len(wires)):
-                    op_list.append(local_field(weights[l][len(wires) + i], wires=wires[i]))
+                    for i in range(len(wires))
```

Comment:
Wasn't aware of this syntax working, though I guess why not! :open_mouth: :100: 

One thing: could we have the generator def in a line first? It could improve readability. E.g., (non-tested)
```py
multirz_group = (qml.MultiRZ(
                        weights[l][i], wires=wires.subset([i, i + 1], periodic_boundary=True)
                    )
                    for i in range(len(wires)))
op_list.extend(multirz_group)
```